clippy_lints/src/blacklisted_name.rs

```diff
@@ -37,6 +38,24 @@ impl_lint_pass!(BlacklistedName => [BLACKLISTED_NAME]);
 
 impl<'tcx> LateLintPass<'tcx> for BlacklistedName {
     fn check_pat(&mut self, cx: &LateContext<'tcx>, pat: &'tcx Pat<'_>) {
+        // Check whether we are under the `test` attribute.
+        if let Some(enclosing) = cx.enclosing_body {
```

Comment:
I don't think this works. E.g.:

```rust
#[test]
fn test_outer() {
    fn no_test_inner() {
        let foo = 42;
    }
    no_test_inner();
}
```

This will probably lint, since the enclosing body of `let foo ...` is `no_test_inner` which doesn't have the `#[test]` attribute.